src/commands/query/info.cc

```diff
@@ -127,7 +127,7 @@ int InfoCmd::execute( Zypper &zypper, const std::vector<std::string> &positional
   }
 
   printInfo( zypper, positionalArgs_r, _options );
-  return ZYPPER_EXIT_OK;
+  return zypper.exitInfoCode();
```

Comment:
Actually this should not be needed. [`main()`](https://github.com/openSUSE/zypper/blob/master/src/main.cc#L162) checks whether zypper would return EXIT_OK, and if it would it returns any pending ExitInfo code instead. Internally  ExitInfo should not be treated as an error nor should it disguise any _real_ errors. That's why they are handles at the end.

Reply:
Fixed.